### PR TITLE
Add AWS SSO terraform modules

### PR DESCRIPTION
**Pull request by rifisdfds**
This PR adds the required terraform modules for our AWS SSO implementation

### CODE REVIEW

**Comment by emil2k** [comment]
@rifisdfds What is the recommended way of testing these modules? Can we create another instance of the AWS SSO or is there one per management account?

**Comment by rifisdfds** [comment]
@emil2k One per management account. The tests I've performed with this module is to run the pre-prime container and run a `terragrunt plan' from master-account/eu-west-1/iam-identity-center in the branch here https://github.com/dfds/pre-prime/pull/76 pointing towards this branch in infrastructure-modules

I felt it not worth the effort to spin up a brand new management account, aws orgs and pre-prime setup.

**Comment by emil2k** [comment]
I tried this out today:

```
root@65ab31dc2ea3:/dfds/pre-prime/master-account/eu-west-1/iam-identity-center# terragrunt plan --terragrunt-source file:///dfds/infrastructure-modules//security/iam-roles-master
Initializing modules...
- iam_policies in ../../_sub/security/iam-policies

Initializing the backend...

Successfully configured the backend "s3"! Terraform will automatically
use this backend unless the backend configuration changes.

Initializing provider plugins...
- Finding hashicorp/aws versions matching "~> 4.45.0"...
- Installing hashicorp/aws v4.45.0...
- Installed hashicorp/aws v4.45.0 (signed by HashiCorp)

Terraform has created a lock file .terraform.lock.hcl to record the provider
selections it made above. Include this file in your version control repository
so that Terraform can guarantee to make the same selections by default when
you run "terraform init" in the future.

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
var.create_org_account_iam_policy_name
  Policy granting access to create new AWS Organization accounts

  Enter a value: CreateOrgAccount

var.iam_role_description
  Description for IAM role

  Enter a value: A test description

var.iam_role_name
  Name of IAM role

  Enter a value: Identity

var.iam_role_trusted_account_root_arn
  The ARN of the account trusted to assume the role

  Enter a value:

module.iam_policies.data.aws_iam_policy_document.admin: Reading...
module.iam_policies.data.aws_iam_policy_document.create_s3_bucket: Reading...
module.iam_policies.data.aws_iam_policy_document.auditor: Reading...
module.iam_policies.data.aws_iam_policy_document.access_cloudwatchlogs_capability: Reading...
module.iam_policies.data.aws_iam_policy_document.create_org_account: Reading...
module.iam_policies.data.aws_iam_policy_document.capability_access_shared: Reading...
module.iam_policies.data.aws_iam_policy_document.admin: Read complete after 0s [id=618086828]
module.iam_policies.data.aws_iam_policy_document.create_org_account: Read complete after 0s [id=489813301]
module.iam_policies.data.aws_iam_policy_document.access_cloudwatchlogs_devops: Reading...
module.iam_policies.data.aws_iam_policy_document.push_to_ecr: Reading...
module.iam_policies.data.aws_iam_policy_document.create_s3_bucket: Read complete after 0s [id=3136582760]
module.iam_policies.data.aws_iam_policy_document.trusted_account[0]: Reading...
module.iam_policies.data.aws_iam_policy_document.cloudengineer: Reading...
module.iam_policies.data.aws_iam_policy_document.push_to_ecr: Read complete after 0s [id=2227782623]
module.iam_policies.data.aws_iam_policy_document.auditor: Read complete after 0s [id=4114710828]
module.iam_policies.data.aws_iam_policy_document.capability_access_shared: Read complete after 0s [id=3144603532]
module.iam_policies.data.aws_iam_policy_document.access_cloudwatchlogs_capability: Read complete after 0s [id=988835492]
module.iam_policies.data.aws_iam_policy_document.access_cloudwatchlogs_devops: Read complete after 0s [id=911330095]
module.iam_policies.data.aws_iam_policy_document.assume_noncore_accounts: Reading...
module.iam_policies.data.aws_iam_policy_document.cloudengineer: Read complete after 0s [id=329043695]
module.iam_policies.data.aws_iam_policy_document.create_route53_zone: Reading...
module.iam_policies.data.aws_iam_policy_document.trusted_account[0]: Read complete after 0s [id=3598177540]
module.iam_policies.data.aws_iam_policy_document.create_route53_zone: Read complete after 0s [id=3214684367]
module.iam_policies.data.aws_iam_policy_document.assume_noncore_accounts: Read complete after 0s [id=2951280726]

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_iam_role.master_role will be created
  + resource "aws_iam_role" "master_role" {
      + arn                   = (known after apply)
      + assume_role_policy    = jsonencode(
            {
              + Statement = [
                  + {
                      + Action    = "sts:AssumeRole"
                      + Effect    = "Allow"
                      + Principal = {
                          + AWS = ""
                        }
                      + Sid       = ""
                    },
                ]
              + Version   = "2012-10-17"
            }
        )
      + create_date           = (known after apply)
      + description           = "A test description"
      + force_detach_policies = false
      + id                    = (known after apply)
      + managed_policy_arns   = (known after apply)
      + max_session_duration  = 3600
      + name                  = "Identity"
      + name_prefix           = (known after apply)
      + path                  = "/"
      + tags_all              = (known after apply)
      + unique_id             = (known after apply)

      + inline_policy {
          + name   = (known after apply)
          + policy = (known after apply)
        }
    }

  # aws_iam_role_policy.create_org_account will be created
  + resource "aws_iam_role_policy" "create_org_account" {
      + id     = (known after apply)
      + name   = "CreateOrgAccount"
      + policy = jsonencode(
            {
              + Statement = [
                  + {
                      + Action   = [
                          + "organizations:MoveAccount",
                          + "organizations:ListTagsForResource",
                          + "organizations:ListRoots",
                          + "organizations:ListParents",
                          + "organizations:ListAccountsForParent",
                          + "organizations:ListAccounts",
                          + "organizations:DescribeCreateAccountStatus",
                          + "organizations:DescribeAccount",
                          + "organizations:CreateAccount",
                        ]
                      + Effect   = "Allow"
                      + Resource = "*"
                      + Sid      = "OrgCreateAccount"
                    },
                ]
              + Version   = "2012-10-17"
            }
        )
      + role   = (known after apply)
    }

Plan: 2 to add, 0 to change, 0 to destroy.
╷
│ Warning: Value for undeclared variable
│
│ The root module does not declare a variable named "auditor_iam_role_name"
│ but a value was found in file
│ "/dfds/pre-prime/master-account/account.tfvars". If you meant to use this
│ value, add a "variable" block to the configuration.
│
│ To silence these warnings, use TF_VAR_... environment variables to provide
│ certain "global" settings to all configurations in your organization. To
│ reduce the verbosity of these warnings, use the -compact-warnings option.
╵
╷
│ Warning: Value for undeclared variable
│
│ The root module does not declare a variable named
│ "cloudadmin_iam_role_name" but a value was found in file
│ "/dfds/pre-prime/master-account/account.tfvars". If you meant to use this
│ value, add a "variable" block to the configuration.
│
│ To silence these warnings, use TF_VAR_... environment variables to provide
│ certain "global" settings to all configurations in your organization. To
│ reduce the verbosity of these warnings, use the -compact-warnings option.
╵
╷
│ Warning: Values for undeclared variables
│
│ In addition to the other similar warnings shown, 1 other variable(s)
│ defined without being declared.
╵

─────────────────────────────────────────────────────────────────────────────

Note: You didn't use the -out option to save this plan, so Terraform can't
guarantee to take exactly these actions if you run "terraform apply" now.
Releasing state lock. This may take a few moments...
```

The HCL is still a draft, but I guess it still needs some variables to be set.

File: _sub/security/iam-identity-center/main.tf [hunk 1]
```diff
@@ -0,0 +1,31 @@
+data "aws_ssoadmin_instances" "dfds" {}
+
+resource "aws_ssoadmin_permission_set" "capabilityaccess" {
```
Comment (emil2k) [inline _sub/security/iam-identity-center/main.tf:3]: As we will be associating this with capability AWS accounts, would it make sense to have this permission set's ARN as an output?

**Review by emil2k — APPROVED** [review]

**Pull request merged**